### PR TITLE
feat: archiving + archived classrooms

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Archive Classroom](https://www.notion.so/uwblueprintexecs/Archive-Classroom-9a8cfb9cdbd645a7a5781eea9cfcd845)
[Archived tab in Classrooms page](https://www.notion.so/uwblueprintexecs/Archived-tab-in-Classrooms-page-7a13f922f7734c84bcbb505ff0e91639)

## Implementation description
- Show archive option on class cards
- Fix archive modal closing too early
- Show archived classrooms in a separate tab

image
image

## Steps to test
1. Archive an existing classroom.
2. Verify it shows up on the correct page.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx [hunk 1]
```diff
@@ -14,50 +18,76 @@ import DeleteClassroomModal from "./DeleteClassroomModal";
...
+  isArchived,
 }: ClassroomPopoverProps): React.ReactElement => {
   const {
     isOpen: isPopoverOpen,
     onOpen: onPopoverOpen,
     onClose: onPopoverClose,
   } = useDisclosure();
-  const [showDeleteModal, setShowDeleteModal] = React.useState(false);
-  const [showEditModal, setShowEditModal] = React.useState(false);
+  const [showEditModal, setShowEditModal] = useState(false);
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx]: should we use `useDisclosure`? i think for the most part i updated our codebase to use that now

File: frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx [hunk 2]
```diff
@@ -77,6 +107,18 @@ const ClassroomPopover = ({
...
+          {!isArchived && (
+            <>
+              <PopoverButton
+                name="Archive"
+                onClick={() => {
+                  onPopoverClose();
+                  setShowArchiveModal(true);
+                }}
+              />
+              <Divider />
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx]: can we use `<VStack divider={<Divider />}` instead?

File: frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx [hunk 3]
```diff
@@ -14,50 +18,76 @@ import DeleteClassroomModal from "./DeleteClassroomModal";
...
-        {
-          query: GET_CLASSES_BY_TEACHER,
-          variables: { teacherId },
-        },
-      ],
-    },
-  );
+  const [archiveClass] = useMutation<{ deleteClass: string }>(ARCHIVE_CLASS, {
+    variables: { classId },
+    refetchQueries: [
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx]: does `getQueryName` work here?

File: frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx [hunk 4]
```diff
@@ -77,6 +107,18 @@ const ClassroomPopover = ({
             }}
           />
           <Divider />
+          {!isArchived && (
+            <>
+              <PopoverButton
+                name="Archive"
+                onClick={() => {
+                  onPopoverClose();
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/classroom-summary/ClassroomPopover.tsx]:
```suggestion
```
i think these close once the modal opens so we don't need to explicitly call

**Review by carissa-tang — APPROVED** [review]
wow we have feature flags 🤩 it's time to do some A/B testing

**Comment by jfdoming** [comment]
unfortunately our feature flags live in the bundle so we can't exactly switch them on/off LOL, but yes something like it

**Pull request merged**